### ISSUE TITLE
The overwrite flag for FileDialog with save option does not working.

### ISSUE THREAD

**Issue by balokir** [eclipse-platform/eclipse.platform.swt#197]
**Describe the bug**
The FileDialog with save option  still shows a confirmation to overwrite the file even though dialog.setOverwrite(false) is called.

**To Reproduce**
```
			FileDialog dialog = new FileDialog(shell, SWT.SAVE);
			dialog.setOverwrite(false);

			dialog.setText("Save As");
			result = dialog.open();
```

**Expected behavior**
The FileDialog with save option must not show a confirmation to overwrite the file if dialog.setOverwrite(false) is called.

**Environment:**
- Select the platforms on which the behavior is seen.
[ ] All OS
[x] Windows
[ ] Linux
[ ] macOS
 
 - Additional OS info (e.g. OS version, Linux Desktop, etc)
 Windows 7

 - JRE/JDK version
Open JDK 11
 
**Version since**
Eclipse 2021-06 (4.20.0)

**Workaround (or) Additional context**
The code in FileDialog.open() 
```
	fileDialog.GetOptions(options);
	options[0] |= OS.FOS_FORCEFILESYSTEM | OS.FOS_NOCHANGEDIR;
	if ((style & SWT.SAVE) != 0) {
		if (overwrite) options[0] |= OS.FOS_OVERWRITEPROMPT;
```
fileDialog.GetOptions(options) returns flags which contains OS.FOS_OVERWRITEPROMPT bit.

**Comment by Phillipus** [comment]
Which version of Eclipse are you using? IIRC it got fixed in 4.21. It is certainly working for me using 4.23.

**Comment by niraj-modi** [comment]
With my quick test FileDialog#setOverwrite(false) seems to be working as per JavaDoc, try running this modified Snippet72:
<details>
<summary>Modified Snippet72</summary>

```java
/*******************************************************************************
 * Copyright (c) 2000, 2021 IBM Corporation and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.swt.snippets;

/*
 * FileDialog example snippet: prompt for a file name (to save)
 *
 * For a list of all SWT example snippets see
 * http://www.eclipse.org/swt/snippets/
 */
import org.eclipse.swt.*;
import org.eclipse.swt.widgets.*;

public class Snippet72 {

public static void main (String [] args) {
	Display display = new Display ();
	Shell shell = new Shell (display);
	shell.setText("Snippet 72");
	shell.open ();
	FileDialog dialog = new FileDialog (shell, SWT.SAVE);
	String [] filterNames = new String [] {"Image Files", "All Files (*)"};
	String [] filterExtensions = new String [] {"*.gif;*.png;*.xpm;*.jpg;*.jpeg;*.tiff", "*"};
	String filterPath = "/";
	String platform = SWT.getPlatform();
	if (platform.equals("win32")) {
		filterNames = new String [] {"All Files (*)"};
		filterExtensions = new String [] {"*"};
		filterPath = "c:\\";
	}
	dialog.setOverwrite(false);
	dialog.setFilterNames (filterNames);
	dialog.setFilterExtensions (filterExtensions);
	dialog.setFilterPath (filterPath);
	dialog.setFileName ("myfile");
	System.out.println ("Save to: " + dialog.open ());
	while (!shell.isDisposed ()) {
		if (!display.readAndDispatch ()) display.sleep ();
	}
	display.dispose ();
}
}

```

</details>

Not sure if i am missing something adding @nnemkin

**Comment by balokir** [comment]
So it is broken in 4.20 and fixed in  4.21. I'm sorry, I could not find bug in history. I think it must be closed as duplicate.

**Comment by mickaelistria** [comment]
Duplicates https://bugs.eclipse.org/bugs/show_bug.cgi?id=574915
